CONTRIBUTING.md

```diff
@@ -0,0 +1,168 @@
+
+Although we really appreciate feedback and ideas, we will close feature requests that we don't expect to fulfill in the next year.
+
+Many of those ideas etc do have a place on the [forums](https://www.tt-forums.net); and if enough people like it, someone will stand up and make it.
+
+It's usually best discuss in [irc](https://wiki.openttd.org/IRC_channel) before opening a feature request or working on a large feature in a fork.
+Discussion in irc can take time, but it can be productive and avoid disappointment :)
+
+## Pull requests
+
+Good pull requests—patches, improvements, new features—are a fantastic help.
+They should remain focused in scope and avoid containing unrelated commits.
+
+**Please ask first** before embarking on any significant pull request (e.g. implementing features, refactoring code, porting to a different language), otherwise you risk spending a lot of time working on something that the project's developers might not want to merge into the project.
+
+Please adhere to the [coding guidelines](#code-guidelines) used throughout the project (indentation, accurate comments, etc.) and any other requirements (such as test coverage).
+
+Adhering to the following process is the best way to get your work included in the project:
+
+1. [Fork](https://help.github.com/fork-a-repo/) the project, clone your fork, and configure the remotes:
+
+```bash
+git clone https://github.com/<your-username>/OpenTTD.git openttd
+git clone https://github.com/OpenTTD/OpenTTD-git-hooks.git openttd_hooks
+cd openttd
+git remote add upstream https://github.com/OpenTTD/OpenTTD.git
+cd .git/hooks
+ln -s -t . ../../../openttd_hooks/hooks/*
+```
+
+2. If you cloned a while ago, get the latest changes from upstream:
+
+```bash
+git fetch upstream
+```
+
+3. Create a new topic branch (off the main project development branch) to
+contain your feature, change, or fix:
+
+```bash
+git checkout upstream/master -b <topic-branch-name>
+```
+
+4. Commit your changes in logical chunks. Please adhere to these [git commit message guidelines](https://wiki.openttd.org/Commit_style#Commit_message) or your code is unlikely to be merged into the main project.
+Use Git's [interactive rebase](https://help.github.com/articles/interactive-rebase) feature to tidy up your commits before making them public.
+
+5. Locally merge (or rebase) the upstream development branch into your topic branch:
+
+```bash
+git fetch upstream
+git rebase upstream/master
+```
+
+6. Push your topic branch up to your fork the first time:
+
+```bash
+git push --set-upstream origin <topic-branch-name>
+```
+
+And any time after that:
+
+```bash
+git push
+```
+
+7. [Open a Pull Request](https://help.github.com/articles/using-pull-requests/) with a clear title and description against the `master` branch.
+
+**IMPORTANT**: By submitting a patch, you agree to allow the project owners to license your work under the terms of the [GPL v2](https://github.com/OpenTTD/OpenTTD/blob/master/COPYING).
```

Comment:
I really like the License part lower; this sentence feels a bit weird. Possibly just tell them to read the license part? :D